### PR TITLE
Implement IVisualTreeElement for TableView

### LINKED ISSUE #2124: [Bug] TableView and its underlying elements need to implement IVisualTreeElement overrides
### Description

image

When creating a TableView in MAUI, the inner children elements are handled with an internal structure that doesn't use `LogicalChildren`. As a result, when you view the Live Visual Tree or Hot Reload any elements contained within it, it most likely won't work since those items are not being tracked.

In order for it to work properly, we need to implement `IVisualTreeElement` on TableView and its related elements that don't use `LogicalChildren` to get the current visual children for where its items are stored. That should enable proper tracking for TableView.

### PR DESCRIPTION
**Pull request by drasticactions**
This should enable XAML Hot Reload and Live Visual Tree functionality for TableView and its underlying types.

Fixes https://github.com/dotnet/maui/issues/2124

https://user-images.githubusercontent.com/898335/130701356-52574300-fc04-4b41-8249-69b5ae9a98db.mp4

### CODE REVIEW

**Review by mattleibow — APPROVED** [review]

**Comment by mattleibow** [comment]
Good to see us bring hot reload support to the most hardy of controls! MT.D anyone?

File: src/Controls/src/Core/TableSection.cs [hunk 1]
```diff
@@ -155,6 +155,8 @@ void OnChildrenChanged(object sender, NotifyCollectionChangedEventArgs notifyCol
 		}
 
 		IReadOnlyList<Maui.IVisualTreeElement> IVisualTreeElement.GetVisualChildren() => this._children.Cast<IVisualTreeElement>().ToList().AsReadOnly();
+
+		IVisualTreeElement IVisualTreeElement.GetVisualParent() => null;
```
Comment (drasticactions) [inline src/Controls/src/Core/TableSection.cs:159]: We don't have direct access to the parent `TableRoot` or other `TableSection`s here. The other issue for Parent is that the underlying Cells all have Parents of the upper most 'TableView', so the _really_ implement it right, we would need to nest the calls all the way down for the right parents... but Cell is an `Element` and that `Parent` has to be `Element`, which the TableSection's are not. 

For Hot Reload and LVT, we don't need to call GetVisualParent for the section, so leaving it null is fine.

**Pull request merged**